diktat-rules/src/main/kotlin/com/saveourtool/diktat/ruleset/rules/chapter2/kdoc/KdocComments.kt

```diff
@@ -102,28 +102,34 @@ class KdocComments(configRules: List<RulesConfig>) : DiktatRule(
         }
     }
 
-    private fun checkParameterList(node: ASTNode?) {
-        val kdocBeforeClass = node
-            ?.parent { it.elementType == CLASS }
-            ?.findChildByType(KDOC) ?: return
+    private fun checkParameterList(classNode: ASTNode, typeParameterListNode: ASTNode?, valueParameterListNode: ASTNode?) {
+        if (typeParameterListNode == null && valueParameterListNode == null) {
+            return
+        }
+
+        val kdocBeforeClass = classNode.findChildByType(KDOC)
+            ?: return
 
         val parametersInKdoc = kdocBeforeClass
             .kDocTags()
             .filter { it.knownTag == KDocKnownTag.PROPERTY || it.knownTag == KDocKnownTag.PARAM }
 
-        val parametersInConstructor = node
-            .findChildrenMatching { it.elementType == VALUE_PARAMETER }
-            .mapNotNull { it.findChildByType(IDENTIFIER)?.text }
+        val parametersInTypeParameterList = typeParameterListNode
```

Comment:
I guess it could be extracted into the method

Reply:
Done